lib/pangea/extensions/pangea_room_extension.dart

```diff
@@ -808,6 +812,85 @@ extension PangeaRoom on Room {
     );
   }
 
+  Future<void> archive() async {
+    final participants = await requestParticipants();
+    final students = participants
+        .where(
+          (e) =>
+              e.powerLevel < ClassDefaultValues.powerLevelOfAdmin &&
+              e.id != BotName.byEnvironment,
+        )
+        .toList();
+    for (final student in students) {
```

Comment:
Since this function is a little bit error prone, can you wrap this is a try catch with some error logging?